### PR TITLE
coreos-koji-tagger: support yaml lockfiles

### PR DESCRIPTION
**Pull request by dustymabe**
- coreos-koji-tagger: update instructions for testing in stage
- coreos-koji-tagger: rework test setup using Mock
- coreos-koji-tagger: support yaml lockfiles

### CODE REVIEW

**Comment by dustymabe** [comment]
this builds on #58 - please see individual commit messages and review only the last 3 commits

File: coreos-koji-tagger/coreos_koji_tagger.py [hunk 1]
```diff
@@ -305,15 +305,18 @@ def process_lockfiles(self, commit=None):
...
+                    logger.info(f'Attempting to retrieve data from {url}')
+                    r = requests.get(url)
+                    if r.ok:
+                        # parse the lockfile and add the set of rpm NEVRAs (strings)
+                        desiredrpms.update(parse_lockfile_data(r.text, filetype))
+                        break # If both yaml and json files exist, only parse one 
+                              # of them. Prefer yaml.
+                    else:
+                        # Log any errors we encounter. 404s are ok, but won't hurt to log
+                        logger.warning('URL request error: %s' % r.text.strip())
```
Comment (jlebon) [inline coreos-koji-tagger/coreos_koji_tagger.py]: Not directly related, though can we do a check on whether `desiredrpms` is empty here and return early if so?

Comment (dustymabe) [reply coreos-koji-tagger/coreos_koji_tagger.py]: IIUC the only reason that `desiredrpms` would be empty is if no lockfiles existed in the repo at all. Does that match your understanding?

If so, is targeting a repo with no lockfiles valid. If valid is it something we want to optimize for?

Comment (jlebon) [reply coreos-koji-tagger/coreos_koji_tagger.py]: I was thinking more about it in terms of reasoning through the code than optimizing for it. Getting the trivial cases out of the way means we don't have to think about them when hacking on code lower down.

Comment (dustymabe) [reply coreos-koji-tagger/coreos_koji_tagger.py]: ok. I'll address in a followup: https://github.com/coreos/fedora-coreos-releng-automation/pull/62

Comment (dustymabe) [reply coreos-koji-tagger/coreos_koji_tagger.py]: Sound good?

**Pull request merged**